planetiler-custommap/src/main/java/com/onthegomap/planetiler/custommap/ConfiguredFeature.java

```diff
@@ -0,0 +1,302 @@
+    Function<SourceFeature, Integer> attributeZoomProducer;
+
+    if (minTileCoverage != null && minTileCoverage > 0.0) {
+      attributeZoomProducer = attributeZoomThreshold(minTileCoverage, minZoom);
+    } else {
+      attributeZoomProducer = sf -> minZoom;
+    }
+
+    return (sf, f) -> {
+      if (attributeTest.test(sf) && attributeExcludeTest.test(sf)) {
+        f.setAttrWithMinzoom(tagKey, attributeValueProducer.apply(sf), attributeZoomProducer.apply(sf));
+      }
+    };
+  }
+
+  /**
+   * Generates a factory method which creates a {@link Feature} from a {@link FeatureCollector} of the appropriate
+   * geometry type.
+   * 
+   * @param layerName - name of the layer
+   * @param type      - type of geometry
+   * @return geometry factory method
+   */
+  private static Function<FeatureCollector, Feature> geometryMapFeature(String layerName, FeatureGeometryType type) {
+    switch (type) {
+      case polygon:
+        return fc -> fc.polygon(layerName);
+      case linestring:
+        return fc -> fc.line(layerName);
+      case point:
+        return fc -> fc.point(layerName);
+      default:
+        throw new IllegalArgumentException("Unhandled geometry type " + type);
+    }
+  }
+
+  /**
+   * Generate a test for whether a source feature is of the correct geometry to be included in the tile.
+   * 
+   * @param type type of geometry
+   * @return geometry test method
+   */
+  private static Predicate<SourceFeature> geometryTest(FeatureGeometryType type) {
+    switch (type) {
+      case polygon:
+        return sf -> sf.canBePolygon();
+      case linestring:
+        return sf -> sf.canBeLine();
+      case point:
+        return sf -> sf.isPoint();
+      default:
+        throw new IllegalArgumentException("Unhandled geometry type " + type);
+    }
+  }
+
+  /**
+   * Test to determine whether to include a source feature in the tiles
+   * 
+   * @param sourceFeature - source feature
+   * @return true if the feature will be rendered
+   */
+  public boolean includeWhen(SourceFeature sourceFeature) {
+    //Is this from the right source?
+    if (!sources.contains(sourceFeature.getSource())) {
+      return false;
+    }
+
+    //Is this the right type of geometry?
+    if (!geometryTest.test(sourceFeature)) {
+      return false;
+    }
+
+    //Check for matching tags
+    return tagTest.test(sourceFeature);
+  }
+
+  /**
+   * Generate a tile feature based on a source feature
+   * 
+   * @param sourceFeature - input source feature
+   * @param features      - output rendered feature collector
+   */
+  public void processFeature(SourceFeature sourceFeature, FeatureCollector features) {
+
+    Feature f = geometryFactory.apply(features);
+
+    f.setBufferPixels(BUFFER_SIZE)
+      // and also whenever you set a label grid size limit, make sure you increase the buffer size so no
+      // label grid squares will be the consistent between adjacent tiles
+      .setBufferPixelOverrides(ZoomFunction.maxZoom(12, 32));
```

Comment:
I don't think we need these yet - the default buffer pixel size is 4, so we'll only need to set this when users can override it at the layer and feature-level. And buffer pixel overrides is only necessary for points using the equivalent of `LabelGrid` from postgis

https://github.com/onthegomap/planetiler/blob/6a7c6791416c06fe6bbbcdebaf74cf078e1c2ea4/planetiler-core/src/main/java/com/onthegomap/planetiler/FeatureCollector.java#L217

Reply:
Fixed in 7cd1efb